hazelcast/src/main/java/com/hazelcast/ringbuffer/ReadResultSet.java

```diff
@@ -49,13 +49,18 @@
 
     /**
      * Return the sequence number for the item at the given index.
-     * The method may throw a {@link UnsupportedOperationException} if there are no
+     * If {@code index >= size() && index < maxSize}, returns {@code 0}.
```

Comment:
Ah, ok, you were requesting sequences for items which didn't exist. How about if we returned `-1` in that case as well? That way we can simply say that the method returns `-1` if there is no sequence (which was the intent of the implementation in the first place).